index.js

```diff
@@ -114,108 +129,125 @@ module.exports = class NoiseSecretStream extends Duplex {
     this._tmp = 1
     this._message = null
 
-    if (this._setup) {
-      if (this._noiseHandshake !== null) {
-        const split = noise.readMessage(this._noiseHandshake, message, EMPTY)
-        if (split) this._onhandshake(split, [])
-        else this._handshakeRT()
-        return
+    if (this._setup === true) {
```

Comment:
If you pass in `opts.tx` (so `setupSecretStream` is called in the constructor), it looks like `this._setup` will still be `true`. That a bug?

Reply:
No, cause it still need to send over the secretstream header in the first message (hits the else below)